### PR TITLE
Small python cleanups

### PR DESCRIPTION
**Pull request by eric-wieser**
Some small changes before I add some hacks in my local copy to let me use this on pygae/lean-ga

### CODE REVIEW

**Review by gebner — APPROVED** [review]
Rest looks good to me.

File: print_docs.py [hunk 1]
```diff
@@ -166,9 +160,8 @@ def separate_results(objs):
   return file_map, loc_map
 
 def load_json():
-  f = open('export.json', 'r', encoding='utf-8')
-  decls = json.load(f, strict=False)
-  f.close()
+  with open('export.json', 'r', encoding='utf-8'):
```
Comment (gebner) [inline print_docs.py:163]:
```suggestion
  with open('export.json', 'r', encoding='utf-8') as f:
```

File: print_docs.py [hunk 2]
```diff
@@ -223,26 +216,28 @@ def linkify_type(string):
...
   remove_keep_contents_patterns = [
-    '(?m)^([\s\t]*)([\*\-\+]|\d\.)\s+',
-    '\*\*([^*]+)\*\*',
-    '\*([^*]+)\*',
-    '(?m)^\#{1,6}\s*([^#]+)\s*(\#{1,6})?$',
-    '__([^_]+)__',
-    '_([^_]+)_',
-    '\!\[(.*?)\]\s?[\[\(].*?[\]\)]',
-    '\[(.*?)\][\[\(].*?[\]\)]'
+    r'(?m)^([\s\t]*)([\*\-\+]|\d\.)\s+',
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply print_docs.py:223]: I think this is the only case where that makes a difference. It seems the python regex doesn't care about a literal tab character vs an escape sequence:
```
In [9]: import re

In [10]: re.match('\t', '\t')
Out[10]: <re.Match object; span=(0, 1), match='\t'>

In [11]: re.match(r'\t', '\t')
Out[11]: <re.Match object; span=(0, 1), match='\t'>
```

**Comment by eric-wieser** [comment]
CI passing now, thanks for catching that missing `as f`

**Pull request merged**